### PR TITLE
Add function to fetch all CKComponentContext items.

### PR DESCRIPTION
**Pull request by adamjernst**
This could be used to implement a system that bridges component context into some other language.

### CODE REVIEW

File: ComponentKit/Utilities/CKComponentContextHelper.h [hunk 1]
```diff
@@ -25,3 +25,9 @@ class CKComponentContextHelper {
   template<typename T>
   friend class CKComponentContext;
 };
+
+/**
+ Returns an array with all the items that are currently in CKComponentContext.
+ Avoid using this unless you really know what you're doing.
+ */
+NSArray *CKAllComponentContextItems();
```
Comment (adamjernst) [inline ComponentKit/Utilities/CKComponentContextHelper.h]: I put it in CKComponentContextHelper.h to discourage its use.

Comment (ocrickard) [inline ComponentKit/Utilities/CKComponentContextHelper.h]: Would it be worth explaining what types of things you might use this for beyond the vague warning?

Comment (adamjernst) [inline ComponentKit/Utilities/CKComponentContextHelper.h]: @ocrickard Added explanation in a future commit

File: ComponentKit/Utilities/CKComponentContextHelper.h [hunk 2]
```diff
@@ -16,12 +16,52 @@ struct CKComponentContextPreviousState {
...
+};
+
+/** Internal helper class. Avoid using this externally. */
+struct CKComponentContextHelper {
   static CKComponentContextPreviousState store(id key, id object);
   static void restore(const CKComponentContextPreviousState &storeResult);
   static id fetch(id key);
+  /**
+   Returns an array with all the items that are currently in CKComponentContext.
+   This could be used to bridge CKComponentContext items to another language or system.
```
Comment (ocrickard) [inline ComponentKit/Utilities/CKComponentContextHelper.h]: lol

File: ComponentKitTests/CKComponentContextTests.mm [hunk 3]
```diff
@@ -69,4 +73,81 @@ - (void)testTriplyNestedComponentContextWithNilMiddleValueCorrectlyRestoresOuter
   XCTAssertTrue(CKComponentContext<NSObject>::get() == outer);
 }
 
+- (void)testFetchingAllComponentContextItemsReturnsArray
```
Comment (eczarny) [inline ComponentKitTests/CKComponentContextTests.mm]: `testFetchingAllComponentContextItemsReturnsObjects`? It isn't an array anymore.

**Comment by eczarny** [comment]
Seems reasonable to me. Great comments.

@adamjernst mind checkout out the failing WildeGuess build?

**Pull request merged**